fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java

```diff
@@ -229,71 +212,77 @@ public List<TScanRangeLocations> getScanRangeLocations(long maxScanRangeLength)
-                    partitionMap.put(partition, partitionId);
-                }
+        String catalogName = srIcebergTable.getCatalogName();
+        long snapshotId = snapshot.get().snapshotId();
 
-                TScanRangeLocations scanRangeLocations = new TScanRangeLocations();
-
-                THdfsScanRange hdfsScanRange = new THdfsScanRange();
-                hdfsScanRange.setFull_path(file.path().toString());
-                hdfsScanRange.setOffset(task.start());
-                hdfsScanRange.setLength(task.length());
-                // For iceberg table we do not need partition id
-                hdfsScanRange.setPartition_id(-1);
-                hdfsScanRange.setFile_length(file.fileSizeInBytes());
-                hdfsScanRange.setFile_format(IcebergApiConverter.getHdfsFileFormat(file.format()).toThrift());
-
-                hdfsScanRange.setDelete_files(task.deletes().stream().map(source -> {
-                    TIcebergDeleteFile target = new TIcebergDeleteFile();
-                    target.setFull_path(source.path().toString());
-                    target.setFile_content(
-                            source.content() == FileContent.EQUALITY_DELETES ? TIcebergFileContent.EQUALITY_DELETES :
-                                    TIcebergFileContent.POSITION_DELETES);
-                    target.setLength(source.fileSizeInBytes());
-
-                    if (source.content() == FileContent.EQUALITY_DELETES) {
-                        source.equalityFieldIds().forEach(fieldId -> {
-                            equalityDeleteColumns.add(
-                                    srIcebergTable.getNativeTable().schema().findColumnName(fieldId));
-                        });
-                    }
-
-                    return target;
-                }).collect(Collectors.toList()));
-                TScanRange scanRange = new TScanRange();
-                scanRange.setHdfs_scan_range(hdfsScanRange);
-                scanRangeLocations.setScan_range(scanRange);
-
-                // TODO: get hdfs block location information for scheduling, use iceberg meta cache
-                TScanRangeLocation scanRangeLocation = new TScanRangeLocation(new TNetworkAddress("-1", -1));
-                scanRangeLocations.addToLocations(scanRangeLocation);
-
-                result.add(scanRangeLocations);
-            }
+        List<RemoteFileInfo> splits = GlobalStateMgr.getCurrentState().getMetadataMgr().getRemoteFileInfos(
+                catalogName, srIcebergTable, null, snapshotId, predicate);
+
+        if (splits.isEmpty()) {
+            LOG.warn("There is no scan tasks after planFies on {}.{} and predicate: [{}]",
+                    srIcebergTable.getRemoteDbName(), srIcebergTable.getRemoteTableName(), predicate);
+            return;
         }
 
-        scanNodePredicates.setSelectedPartitionIds(partitionMap.values());
+        RemoteFileDesc remoteFileDesc = splits.get(0).getFiles().get(0);
```

Comment:
A litter wierd,  could one planTask for one RemoteFileDesc？

Reply:
The purpose is to reduce the size of the cache, otherwise the other fields will also have the initial memory